### PR TITLE
Fixes #20242: Move to a new Url Technique api so anyone can create technique like the editor through api

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/20242

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala [hunk 1]
```diff
@@ -357,7 +357,7 @@ class TechniqueSerializer(parameterTypeService: ParameterTypeService) {
     def serializeMethodBlock(block: MethodBlock): JValue = {
       ( ("condition" -> block.condition)
       ~ ("component" -> block.component)
-      ~ ("reportingLogic" -> serializeCompositionRule(block.reportingLogic))
+      ~ ("reporting_logic" -> serializeCompositionRule(block.reportingLogic))
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala:360]: remaining snake case

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala [hunk 2]
```diff
@@ -372,9 +372,9 @@ class TechniqueSerializer(parameterTypeService: ParameterTypeService) {
           )
       }
 
-      ( ("method_name" -> call.methodId.value)
-      ~ ("class_context" -> call.condition)
-      ~ ("disableReporting" -> call.disabledReporting)
+      ( ("method" -> call.methodId.value)
+      ~ ("condition" -> call.condition)
+      ~ ("disable_reporting" -> call.disabledReporting)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala:377]: here too

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala [hunk 3]
```diff
@@ -435,17 +436,24 @@ class TechniqueSerializer(parameterTypeService: ParameterTypeService) {
...
+    ~ ("agents" -> agentSupport)
+    ~ ("parameters" -> parameters)
     ~ ("documentation" -> method.documentation)
-    ~ ("deprecated" -> method.deprecated)
-    ~ ("rename" -> method.renameTo)
-    ~ ("parameter_rename" -> method.renameParam.map(t => ( "old" -> t._1) ~ ("new" -> t._2)))
+    ~ ("deprecated" -> ( method.deprecated match {
+                           case None => None
+                           case Some(info) => Some(( ("info" -> method.classPrefix)
+                                              ~ ("rename" -> method.classParameter.value)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala:452]: shouldn't it be `renamedTo` ? Or `replacedBy` ?

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala [hunk 4]
```diff
@@ -435,17 +436,24 @@ class TechniqueSerializer(parameterTypeService: ParameterTypeService) {
...
+                                              ) )
+                        })
+       )
+    ~ ("renameParameters" -> method.renameParam.map(t => ( "old" -> t._1) ~ ("new" -> t._2)))
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechnique.scala:456]: same as above, I'm not sure about that name

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Review by fanf — APPROVED** [review]

**Pull request merged**